tests/testthat/test-df_mat_mul.R

```diff
@@ -0,0 +1,19 @@
+df <- data.frame(matrix(1:100, ncol = 5))
+mat <- matrix(1:4, ncol = 2)
+df_mat_mul(df, mat, "z", dplyr::num_range("X", 2:3))
+test_that("df_mat_mul checks inputs", {
+  expect_error(df_mat_mul(30,mat))
+  expect_error(df_mat_mul(df,20))
+})
+
+test_that("Incompatible matrix multipication cannot happen", {
+  expect_error(df_mat_mul(df, mat, "z", dplyr::num_range("X", 1:3)))
+})
+
+test_that("Matrix multiplication is being handled as expected", {
+  X <- df[c(1,4,5)]
+  Z <- as.data.frame(as.matrix(df[2:3]) %*% mat)
+  colnames(Z) <- c("z1","z2")
+  output <- cbind(X,Z)
+  expect_identical(df_mat_mul(df,mat, "z", dplyr::num_range("X", 2:3)),output)
+})
```

Comment:
* Maybe also a test to ensure that if we request columns without available names?
* Test to make sure that other `tidyselect` functions work.

Reply:
How would I go about and do this when the third argument takes in a default, but the fourth argument is a ...?
For examples, `df_mat_mul(x,y,1:2)` assumes 1:2 is the third argument and not the ....

Reply:
Sorry, I forgot how this function works. So there's only one task really. What happens if you do:
```r
df_mat_mul(df, mat, "z", dplyr::num_range("Q", 2:3))
```

Reply:
Right, so there should probably be a test that expects an error.